tests/test_date.py

```diff
@@ -84,224 +126,348 @@ def test_date_arguments_and_date_range_with_default_post_days(self):
         param(low=datetime(2014, 4, 15),
               high=datetime(2014, 5, 15),
-              period='month',
+              period_size='month',
               expected_results=[datetime(2014, 4, 1), datetime(2014, 5, 1)]),
         param(low=datetime(2014, 10, 30, 4, 30),
               high=datetime(2014, 11, 7, 5, 20),
-              period='week',
+              period_size='week',
               expected_results=[datetime(2014, 10, 27), datetime(2014, 11, 3)]),
         param(low=datetime(2014, 8, 13, 13, 21),
               high=datetime(2014, 8, 14, 14, 7),
-              period='day',
+              period_size='day',
               expected_results=[datetime(2014, 8, 13), datetime(2014, 8, 14)]),
         param(low=datetime(2014, 5, 11, 22, 4),
               high=datetime(2014, 5, 12, 0, 5),
-              period='hour',
+              period_size='hour',
               expected_results=[datetime(2014, 5, 11, 22, 0),
                                 datetime(2014, 5, 11, 23, 0),
                                 datetime(2014, 5, 12, 0, 0)]),
         param(low=datetime(2014, 4, 25, 11, 11, 11),
               high=datetime(2014, 4, 25, 11, 12, 11),
-              period='minute',
+              period_size='minute',
               expected_results=[datetime(2014, 4, 25, 11, 11, 0),
                                 datetime(2014, 4, 25, 11, 12, 0)]),
         param(low=datetime(2014, 12, 31, 23, 59, 58, 500),
               high=datetime(2014, 12, 31, 23, 59, 59, 600),
-              period='second',
+              period_size='second',
               expected_results=[datetime(2014, 12, 31, 23, 59, 58, 0),
                                 datetime(2014, 12, 31, 23, 59, 59, 0)]),
     ])
-    def test_periods(self, low, high, period, expected_results):
-        results = list(date.get_intersecting_periods(low, high, period=period))
-        self.assertEquals(expected_results, results)
+    def test_periods(self, low, high, period_size, expected_results):
+        self.when_intersecting_period_calculated(low, high, period_size=period_size)
+        self.then_results_are(expected_results)
 
     @parameterized.expand([
-        'years',
-        'months',
-        'days',
-        'hours',
-        'minutes',
-        'seconds',
-        'microseconds',
-        'some_period',
+        param('years'),
+        param('months'),
+        param('days'),
+        param('hours'),
+        param('minutes'),
+        param('seconds'),
+        param('microseconds'),
+        param('some_period'),
     ])
-    def test_should_reject_easily_mistaken_dateutil_arguments(self, invalid_period):
-        low = datetime(2014, 6, 15)
-        period = datetime(2014, 6, 25)
-
-        with self.assertRaisesRegexp(ValueError, "Invalid period: {}".format(invalid_period)):
-            date.get_intersecting_periods(low, period, period=invalid_period).next()
+    def test_should_reject_easily_mistaken_dateutil_arguments(self, period_size):
```

Comment:
I missed the the test case hierarchy - looking at it makes sense. I don't think we need to change the name.